storm-core/src/jvm/backtype/storm/topology/base/BaseWindowedBolt.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package backtype.storm.topology.base;
+
+import backtype.storm.Config;
+import backtype.storm.task.OutputCollector;
+import backtype.storm.task.TopologyContext;
+import backtype.storm.topology.IWindowedBolt;
+import backtype.storm.topology.OutputFieldsDeclarer;
+import backtype.storm.windowing.TupleWindow;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public abstract class BaseWindowedBolt implements IWindowedBolt {
+    private static final Logger LOG = LoggerFactory.getLogger(BaseWindowedBolt.class);
+
+    private final transient Map<String, Object> windowConfiguration;
+
+    /**
+     * Holds a count value for count based windows and sliding intervals.
+     */
+    public static class Count {
+        public final int value;
+
+        public Count(int value) {
+            this.value = value;
+        }
+    }
+
+    /**
+     * Holds a Time duration for time based windows and sliding intervals.
+     */
+    public static class Duration {
```

Comment:
I don't see a lot of value in having a Count or a Duration class, Count especially.  They either wrap holding an int that is quickly unwrapped, or converting a timeUnit to millis, which too is quickly unwrapped.